components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/internal/DashboardMetadataProviderImpl.java

```diff
@@ -238,14 +319,74 @@ public void updateDashboardRoles(String dashboardUrl, Map<String, List<String>>
      * @return
      */
     private List<Permission> buildDashboardPermissions(String dashboardUrl) {
-        String prefix = "dashboard." + dashboardUrl;
         List<Permission> permissions = new ArrayList<>();
-        permissions.add(new Permission(PERMISSION_APP_NAME, prefix + ".owner"));
-        permissions.add(new Permission(PERMISSION_APP_NAME, prefix + ".editor"));
-        permissions.add(new Permission(PERMISSION_APP_NAME, prefix + ".viewer"));
+        permissions.add(new Permission(PERMISSION_APP_NAME, dashboardUrl + OWNER_PERMISSION_SUFFIX));
+        permissions.add(new Permission(PERMISSION_APP_NAME, dashboardUrl + EDITOR_PERMISSION_SUFFIX));
+        permissions.add(new Permission(PERMISSION_APP_NAME, dashboardUrl + VIEWER_PERMISSION_SUFFIX));
         return permissions;
     }
 
+    private boolean hasRoles(String user, Role role) {
+        try {
+            List<org.wso2.carbon.analytics.idp.client.core.models.Role> userRoles = identityClient.getUserRoles(user);
```

Comment:
Why intermediate local variables?? 

Java 8 for the WIN!  :star2: :star2: :star2:
```java
return identityClient.getUserRoles(user).stream()
    .filter(...)
    .findAny()
    .isPresent()
```

Reply:
Fixed